src/rgw/rgw_http_client.cc

```diff
@@ -790,6 +790,18 @@ int RGWHTTPManager::set_threaded()
     ldout(cct, 0) << "ERROR: pipe() returned errno=" << r << dendl;
     return r;
   }
+  r=fcntl(thread_pipe[0],F_SETFL,O_NONBLOCK);
+  if(r<0){
+  	r=-errno;
+  	ldout(cct, 0) << "ERROR: fcntl() returned errno=" << r << dendl;
+  	return r;
+  }
+  r=fcntl(thread_pipe[1],F_SETFL,O_NONBLOCK);
+  if(r<0){
+  	r=-errno;
+  	ldout(cct, 0) << "ERROR: fcntl() returned errno=" << r << dendl;
+  	return r;
+  }
```

Comment:
i think that we only need the read side `thread_pipe[0]` to be non-blocking, can you try without the second call to `fcntl()`?